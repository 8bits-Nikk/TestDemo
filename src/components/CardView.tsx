import React from 'react';
import {StyleSheet, Text, TouchableOpacity, View} from 'react-native';
import {IUsers} from '../service/users/users';

type props = {
  item: IUsers;
  onPress: () => void;
};
const CardView: React.FC<props> = ({item, onPress}) => {
  return (
    <TouchableOpacity
      testID={'card-view'}
      style={style.itemContainer}
      onPress={onPress}>
      <Text style={style.text} numberOfLines={1} testID={'card-name'}>
        {item.name}
      </Text>
      <View style={style.subContainer}>
        <View
          testID={'status-dot'}
          style={[
            style.dot,
            // eslint-disable-next-line react-native/no-inline-styles
            {
              backgroundColor: item.status === 'active' ? '#00ff00' : '#ff0000',
            },
          ]}
        />
        <Text testID={'card-status'}>{item.status}</Text>
      </View>
    </TouchableOpacity>
  );
};

const style = StyleSheet.create({
  itemContainer: {
    flexDirection: 'row',
    backgroundColor: '#fff',
    padding: 16,
    borderRadius: 16,
    marginBottom: 12,
    flex: 1,
    alignItems: 'center',
  },
  text: {
    fontSize: 16,
    flex: 0.8,
  },
  dot: {
    backgroundColor: '#ff0000',
    padding: 6,
    borderRadius: 8,
    marginEnd: 8,
  },
  subContainer: {flexDirection: 'row', alignItems: 'center', flex: 0.2},
});

export default CardView;
