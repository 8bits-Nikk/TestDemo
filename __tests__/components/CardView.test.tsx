import React from 'react';
import {
  cleanup,
  fireEvent,
  render,
  screen,
} from '@testing-library/react-native';
import CardView from '../../src/components/CardView';

describe('CardView Tests', () => {
  const item = {
    id: 1,
    email: 'test@mail.com',
    gender: 'male',
    status: 'active',
    name: 'Test',
  };

  const onPress = jest.fn();

  afterEach(() => {
    cleanup();
  });

  it('Should render correctly', () => {
    const tree = render(<CardView item={item} onPress={onPress} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should render correct name and status', () => {
    render(<CardView item={item} onPress={onPress} />);
    const nameText = screen.queryByTestId('card-name');
    const statusText = screen.queryByTestId('card-status');

    expect(nameText?.props.children).toBe(item.name);
    expect(statusText?.props.children).toBe(item.status);
  });

  it('Should render correct color', () => {
    render(<CardView item={item} onPress={onPress} />);
    const dot = screen.queryByTestId('status-dot');
    const dotColor = dot?.props.style[1].backgroundColor;
    // active => '#00ff00' inactive => '#ff0000'
    expect(dotColor).toBe('#00ff00');
  });

  it('Should run onClick Fun', () => {
    render(<CardView item={item} onPress={onPress} />);
    fireEvent.press(screen.getByTestId('card-view'));
    expect(onPress).toBeCalledTimes(1);
  });
});
