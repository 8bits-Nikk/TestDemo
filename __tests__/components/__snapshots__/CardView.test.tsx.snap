// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardView Tests Should render correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "borderRadius": 16,
      "flex": 1,
      "flexDirection": "row",
      "marginBottom": 12,
      "opacity": 1,
      "padding": 16,
    }
  }
  testID="card-view"
>
  <Text
    numberOfLines={1}
    style={
      Object {
        "flex": 0.8,
        "fontSize": 16,
      }
    }
    testID="card-name"
  >
    Test
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 0.2,
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#ff0000",
            "borderRadius": 8,
            "marginEnd": 8,
            "padding": 6,
          },
          Object {
            "backgroundColor": "#00ff00",
          },
        ]
      }
      testID="status-dot"
    />
    <Text
      testID="card-status"
    >
      active
    </Text>
  </View>
</View>
`;
